show/bgp_common.py

```diff
@@ -0,0 +1,420 @@
+    print("       F - PBR, f - OpenFabric,")
+    print("       > - selected route, * - FIB route, q - queued route, r - rejected route\n")
+
+def show_routes(args, namespace, display, verbose, ipver):
+    import utilities_common.bgp_util as bgp_util
+    """Show IPv4/IPV6 routing table"""
+    if display is None:
+        if multi_asic.is_multi_asic():
+            display = constants.DISPLAY_EXTERNAL
+    else:
+        if multi_asic.is_multi_asic():
+            if display not in multi_asic_util.multi_asic_display_choices():
+                print "dislay option '{}' is not a valid option.".format(display)
+                return
+        else:
+            print "dislay option '{}' is not a valid option.".format(display)
+            return
+    device = multi_asic_util.MultiAsic(display, namespace)
+    arg_strg = ""
+    found_json = 0
+    ns_l = []
+    print_ns_str = False
+    filter_back_end = False
+    filter_by_ip = False
+    asic_cnt = 0
+    if multi_asic.is_multi_asic():
+        '''
+        handling multi-ASIC by gathering the output from specified/all name space into a dictionary via
+        jason option and then filter out the json entries (by removing those next Hop that are
+        back-end ASIC if display is for front-end only). If the entry itself has no more next Hop after filtering,
+        then skip over that particular route entry.  Once completed, if the user chose "json" option,
+        then just print out the dictionary in Json format accordingly. But if no "json" option specified,
+        then print out the header and the decoded entry representation for each route accordingly.
+        if user specified a namespace, then print everything.
+        if user did not specify name space but specified display for all (include backend), then print each namespace
+        without any filtering.  But if display is for front-end only, then do filter and combine all output(merge same
+        routes from all namespace as additional nexthops)
+        This code is based on FRR 7.2 branch. If we moved to a new version we may need to change here as well
+        '''
+        if display != constants.DISPLAY_ALL:
+            filter_back_end = True
+        try:
+            for name_s in device.get_ns_list_based_on_options():
+                ns_l.append(name_s)
+            asic_cnt = len(ns_l)
+            if asic_cnt > 1 and display == constants.DISPLAY_ALL:
+                print_ns_str = True
+        except ValueError:
+            print "namespace '{}' is not valid. valid name spaces are:\n{}".format(namespace, multi_asic_util.multi_asic_ns_choices())
+            return
+    else:
+        if namespace is not None:
+            print "namespace option is not applicable for non-multi-asic platform"
+            return
+        if display is not None:
+            print "display option is not applicable for non-multi-asic platform"
+            return
```

Comment:
The display option argument validation is done on the top of the function. This code is duplicate ?

Reply:
@arlakshm Yes. it got covered at the top already. Will remove it.